File: projects/jiyi/jiyi.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
	"strings"
	"time"
)

// 配置
const (
	JiyiDir      = "/home/admin/Ziwei/jiyi"
	MemoryDir    = JiyiDir + "/memory"
	IndexFile    = JiyiDir + "/index.json"
	Version      = "1.0.0"
)

// 记忆结构
type Memory struct {
	ID        string    `json:"id"`
	Category  string    `json:"category"`
	Content   string    `json:"content"`
	Tags      []string  `json:"tags"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	IsNew     bool      `json:"is_new"`
}

// 索引结构
type Index struct {
	Categories map[string][]string `json:"categories"`
	Tags       map[string][]string `json:"tags"`
	LastUpdate time.Time           `json:"last_update"`
}

// 颜色定义
const (
	ColorReset  = "\033[0m"
	ColorRed    = "\033[31m"
	ColorGreen  = "\033[32m"
	ColorYellow = "\033[33m"
	ColorBlue   = "\033[34m"
	ColorPurple = "\033[35m"
	ColorCyan   = "\033[36m"
)

func main() {
	if len(os.Args) < 2 {
		showHelp()
		return
	}

	command := os.Args[1]

	switch command {
	case "add", "a":
		handleAdd(os.Args[2:])
	case "search", "s", "find":
		handleSearch(os.Args[2:])
	case "list", "ls":
		handleList(os.Args[2:])
	case "categories", "cat":
		handleCategories()
	case "tags":
		handleTags()
	case "version", "v", "-v", "--version":
		showVersion()
	case "help", "h", "-h", "--help":
		showHelp()
	default:
		// 默认搜索
		handleSearch(os.Args[1:])
	}
}

func showHelp() {
	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          jiyi - 紫微智控记忆命令                        ║%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()
	fmt.Printf("%s用法:%s\n", ColorCyan, ColorReset)
	fmt.Println("  jiyi <命令> [参数]")
	fmt.Println()
	fmt.Printf("%s命令:%s\n", ColorCyan, ColorReset)
	fmt.Println("  add, a <内容> [分类] [标签...]  - 添加记忆")
	fmt.Println("  search, s, find <关键词>       - 搜索记忆")
	fmt.Println("  list, ls [分类]                - 列出记忆")
	fmt.Println("  categories, cat                - 显示所有分类")
	fmt.Println("  tags                           - 显示所有标签")
	fmt.Println("  version, v                     - 显示版本")
	fmt.Println("  help, h                        - 显示帮助")
	fmt.Println()
	fmt.Printf("%s示例:%s\n", ColorCyan, ColorReset)
	fmt.Println("  jiyi add \"runtask 命令用于启动任务\" 命令 runtask 自动化")
	fmt.Println("  jiyi search runtask")
	fmt.Println("  jiyi list 命令")
	fmt.Println("  jiyi")
	fmt.Println()
}

func showVersion() {
	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          jiyi - 紫微智控记忆命令                        ║%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()
	fmt.Printf("%s版本:%s v%s\n", ColorCyan, ColorReset, Version)
	fmt.Printf("%s日期:%s 2026-02-28\n", ColorCyan, ColorReset)
	fmt.Printf("%s代号:%s 紫微智控 - 记忆命令\n", ColorCyan, ColorReset)
	fmt.Println()
	fmt.Printf("%s功能:%s\n", ColorCyan, ColorReset)
	fmt.Println("  ✅ 分门别类存储")
	fmt.Println("  ✅ 最小存储空间")
	fmt.Println("  ✅ 快速搜索")
	fmt.Println("  ✅ 自动分类")
	fmt.Println("  ✅ 新旧记忆分离")
	fmt.Println()
}

func initJiyi() {
	// 创建目录
	os.MkdirAll(JiyiDir, 0755)
	os.MkdirAll(MemoryDir, 0755)

	// 初始化索引
	if _, err := os.Stat(IndexFile); os.IsNotExist(err) {
		index := Index{
			Categories: make(map[string][]string),
			Tags:       make(map[string][]string),
			LastUpdate: time.Now(),
		}
		saveIndex(index)
	}
}

func loadIndex() Index {
	data, err := os.ReadFile(IndexFile)
	if err != nil {
		return Index{
			Categories: make(map[string][]string),
			Tags:       make(map[string][]string),
			LastUpdate: time.Now(),
		}
	}

	var index Index
	json.Unmarshal(data, &index)
	return index
}

func saveIndex(index Index) {
	data, _ := json.MarshalIndent(index, "", "  ")
	os.WriteFile(IndexFile, data, 0644)
}

func loadMemory(id string) *Memory {
	filePath := filepath.Join(MemoryDir, id+".json")
	data, err := os.ReadFile(filePath)
	if err != nil {
		return nil
	}

	var memory Memory
	json.Unmarshal(data, &memory)
	return &memory
}

func saveMemory(memory Memory) {
	data, _ := json.MarshalIndent(memory, "", "  ")
	filePath := filepath.Join(MemoryDir, memory.ID+".json")
	os.WriteFile(filePath, data, 0644)
}

func generateID() string {
	return fmt.Sprintf("m_%d", time.Now().UnixNano())
}

func autoCategory(content string) string {
	content = strings.ToLower(content)

	// 自动分类关键词
	categories := map[string][]string{
		"命令":     {"命令", "runtask", "look", "jiyi", "执行"},
		"项目":     {"项目", "仓库", "github", "gitee"},
		"配置":     {"配置", "config", "env", "token"},
		"流程":     {"流程", "步骤", "自动", "同步"},
		"错误":     {"错误", "失败", "问题", "bug"},
		"系统":     {"系统", "服务", "监控", "进程"},
		"文档":     {"文档", "说明", "readme", "sop"},
	}

	for cat, keywords := range categories {
		for _, kw := range keywords {
			if strings.Contains(content, kw) {
				return cat
			}
		}
	}

	return "其他"
}

func handleAdd(args []string) {
	if len(args) < 1 {
		fmt.Printf("%s✗ 请提供记忆内容%s\n", ColorRed, ColorReset)
		fmt.Println("用法：jiyi add <内容> [分类] [标签...]")
		return
	}

	initJiyi()

	content := args[0]
	category := "其他"
	tags := []string{}

	// 解析分类和标签
	if len(args) > 1 {
		category = args[1]
		if len(args) > 2 {
			tags = args[2:]
		}
	} else {
		// 自动分类
		category = autoCategory(content)
	}

	memory := Memory{
		ID:        generateID(),
		Category:  category,
		Content:   content,
		Tags:      tags,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
		IsNew:     true,
	}

	saveMemory(memory)

	// 更新索引
	index := loadIndex()
	index.Categories[category] = append(index.Categories[category], memory.ID)
	for _, tag := range tags {
		index.Tags[tag] = append(index.Tags[tag], memory.ID)
	}
	index.LastUpdate = time.Now()
	saveIndex(index)

	fmt.Printf("%s✓ 记忆已添加%s\n", ColorGreen, ColorReset)
	fmt.Printf("  ID: %s\n", memory.ID)
	fmt.Printf("  分类：%s\n", memory.Category)
	fmt.Printf("  标签：%v\n", memory.Tags)
	fmt.Printf("  时间：%s\n", memory.CreatedAt.Format("2006-01-02 15:04:05"))
}

func handleSearch(args []string) {
	if len(args) < 1 {
		// 无参数进入交互模式
		interactiveSearch()
		return
	}

	initJiyi()

	keyword := strings.Join(args, " ")
	keyword = strings.ToLower(keyword)

	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          搜索结果：%s%-40s%s%s\n", ColorBlue, ColorCyan, keyword, ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()

	count := 0
	index := loadIndex()

	// 搜索所有记忆
	for _, ids := range index.Categories {
		for _, id := range ids {
			memory := loadMemory(id)
			if memory == nil {
				continue
			}

			content := strings.ToLower(memory.Content)
			tags := strings.ToLower(strings.Join(memory.Tags, " "))
			category := strings.ToLower(memory.Category)

			if strings.Contains(content, keyword) ||
				strings.Contains(tags, keyword) ||
				strings.Contains(category, keyword) {

				count++
				printMemory(memory, count)
			}
		}
	}

	if count == 0 {
		fmt.Printf("%s✗ 未找到相关记忆%s\n", ColorYellow, ColorReset)
		fmt.Println()
		fmt.Printf("%s建议:%s\n", ColorCyan, ColorReset)
		fmt.Printf("  1. 检查关键词是否正确\n")
		fmt.Printf("  2. 尝试其他关键词\n")
		fmt.Printf("  3. 添加新记忆：jiyi add \"内容\" 分类 标签\n")
		fmt.Println()

		// 询问是否添加新记忆
		fmt.Printf("%s是否添加为新记忆？(y/n): %s", ColorCyan, ColorReset)
		reader := bufio.NewReader(os.Stdin)
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)

		if input == "y" || input == "Y" {
			fmt.Printf("输入记忆内容：")
			content, _ := reader.ReadString('\n')
			content = strings.TrimSpace(content)

			if content != "" {
				// 自动分类并添加
				category := autoCategory(content)
				fmt.Printf("自动分类：%s\n", category)
				fmt.Printf("输入标签（空格分隔）：")
				tagInput, _ := reader.ReadString('\n')
				tags := strings.Fields(strings.TrimSpace(tagInput))

				// 调用 add
				args := append([]string{content, category}, tags...)
				handleAdd(args)
			}
		}
	} else {
		fmt.Printf("%s找到 %d 条记忆%s\n", ColorGreen, count, ColorReset)
	}
}

func interactiveSearch() {
	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          jiyi - 交互式搜索                             ║%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()
	fmt.Printf("%s输入关键词搜索（输入 q 退出）:%s\n", ColorCyan, ColorReset)

	reader := bufio.NewReader(os.Stdin)

	for {
		fmt.Printf("\n%s> %s", ColorGreen, ColorReset)
		input, err := reader.ReadString('\n')
		if err != nil {
			break
		}

		input = strings.TrimSpace(input)
		if input == "q" || input == "quit" {
			break
		}

		if input != "" {
			handleSearch([]string{input})
		}
	}
}

func handleList(args []string) {
	initJiyi()

	category := ""
	if len(args) > 0 {
		category = args[0]
	}

	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	if category != "" {
		fmt.Printf("%s║          记忆列表 - %s%-34s%s%s\n", ColorBlue, ColorCyan, category, ColorBlue, ColorReset)
	} else {
		fmt.Printf("%s║          记忆列表 - 全部%s%15s%s\n", ColorBlue, ColorCyan, ColorBlue, ColorReset)
	}
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()

	index := loadIndex()
	count := 0

	if category != "" {
		// 列出指定分类
		ids, ok := index.Categories[category]
		if !ok {
			fmt.Printf("%s✗ 分类不存在：%s%s\n", ColorRed, ColorCyan, category, ColorReset)
			return
		}

		for _, id := range ids {
			memory := loadMemory(id)
			if memory != nil {
				count++
				printMemory(memory, count)
			}
		}
	} else {
		// 列出所有
		for cat, ids := range index.Categories {
			fmt.Printf("%s【%s】%s\n", ColorPurple, cat, ColorReset)

			for _, id := range ids {
				memory := loadMemory(id)
				if memory != nil {
					count++
					fmt.Printf("  %s\n", memory.Content)
					if memory.IsNew {
						fmt.Printf("    %s[NEW]%s\n", ColorGreen, ColorReset)
					}
					fmt.Println()
				}
			}
		}
	}

	if count == 0 {
		fmt.Printf("%s✗ 没有记忆%s\n", ColorYellow, ColorReset)
	} else {
		fmt.Printf("%s共 %d 条记忆%s\n", ColorGreen, count, ColorReset)
	}
}

func handleCategories() {
	initJiyi()

	index := loadIndex()

	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          记忆分类                                      ║%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()

	for cat, ids := range index.Categories {
		fmt.Printf("  %s%-15s%s %d 条记忆\n", ColorPurple, cat, ColorReset, len(ids))
	}

	fmt.Println()
	fmt.Printf("总计：%d 个分类\n", len(index.Categories))
}

func handleTags() {
	initJiyi()

	index := loadIndex()

	fmt.Printf("%s╔════════════════════════════════════════════════════════╗%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s║          记忆标签                                      ║%s\n", ColorBlue, ColorReset)
	fmt.Printf("%s╚════════════════════════════════════════════════════════╝%s\n", ColorBlue, ColorReset)
	fmt.Println()

	for tag, ids := range index.Tags {
		fmt.Printf("  %s#%-15s%s %d 条记忆\n", ColorCyan, tag, ColorReset, len(ids))
	}

	fmt.Println()
	fmt.Printf("总计：%d 个标签\n", len(index.Tags))
}

func printMemory(memory Memory, index int) {
	fmt.Printf("%s【记忆 #%d】%s\n", ColorPurple, index, ColorReset)
	fmt.Printf("  内容：%s\n", memory.Content)
	fmt.Printf("  分类：%s\n", memory.Category)
	if len(memory.Tags) > 0 {
		fmt.Printf("  标签：%v\n", memory.Tags)
	}
	fmt.Printf("  时间：%s\n", memory.CreatedAt.Format("2006-01-02 15:04:05"))
	if memory.IsNew {
		fmt.Printf("  %s[新记忆]%s\n", ColorGreen, ColorReset)
	}
	fmt.Printf("  文件：%s.json\n", memory.ID)
	fmt.Println()
}

File: projects/jiyi/jiyi.py
#!/usr/bin/env python3
# =============================================================================
# jiyi - 紫微智控记忆命令（Python 版本）
# 功能：分门别类、最小存储、快速搜索
# =============================================================================

import os
import sys
import json
import time
from datetime import datetime

# 配置
JIYI_DIR = "/home/admin/Ziwei/jiyi"
MEMORY_DIR = os.path.join(JIYI_DIR, "memory")
INDEX_FILE = os.path.join(JIYI_DIR, "index.json")
VERSION = "1.0.0"

# 颜色定义
class Colors:
    RESET = "\033[0m"
    RED = "\033[31m"
    GREEN = "\033[32m"
    YELLOW = "\033[33m"
    BLUE = "\033[34m"
    PURPLE = "\033[35m"
    CYAN = "\033[36m"

def init_jiyi():
    """初始化记忆目录"""
    os.makedirs(JIYI_DIR, exist_ok=True)
    os.makedirs(MEMORY_DIR, exist_ok=True)
    
    if not os.path.exists(INDEX_FILE):
        index = {
            "categories": {},
            "tags": {},
            "last_update": datetime.now().isoformat()
        }
        save_index(index)

def load_index():
    """加载索引"""
    try:
        with open(INDEX_FILE, 'r', encoding='utf-8') as f:
            return json.load(f)
    except:
        return {"categories": {}, "tags": {}, "last_update": datetime.now().isoformat()}

def save_index(index):
    """保存索引"""
    with open(INDEX_FILE, 'w', encoding='utf-8') as f:
        json.dump(index, f, ensure_ascii=False, indent=2)

def load_memory(memory_id):
    """加载记忆"""
    try:
        with open(os.path.join(MEMORY_DIR, f"{memory_id}.json"), 'r', encoding='utf-8') as f:
            return json.load(f)
    except:
        return None

def save_memory(memory):
    """保存记忆"""
    with open(os.path.join(MEMORY_DIR, f"{memory['id']}.json"), 'w', encoding='utf-8') as f:
        json.dump(memory, f, ensure_ascii=False, indent=2)

def generate_id():
    """生成记忆 ID"""
    return f"m_{int(time.time() * 1000000)}"

def auto_category(content):
    """自动分类"""
    content_lower = content.lower()
    
    categories = {
        "命令": ["命令", "runtask", "look", "jiyi", "执行"],
        "项目": ["项目", "仓库", "github", "gitee"],
        "配置": ["配置", "config", "env", "token"],
        "流程": ["流程", "步骤", "自动", "同步"],
        "错误": ["错误", "失败", "问题", "bug"],
        "系统": ["系统", "服务", "监控", "进程"],
        "文档": ["文档", "说明", "readme", "sop"],
    }
    
    for cat, keywords in categories.items():
        for kw in keywords:
            if kw.lower() in content_lower:
                return cat
    
    return "其他"

def add_memory(args):
    """添加记忆"""
    if len(args) < 1:
        print(f"{Colors.RED}✗ 请提供记忆内容{Colors.RESET}")
        print("用法：jiyi add <内容> [分类] [标签...]")
        return
    
    init_jiyi()
    
    content = args[0]
    category = args[1] if len(args) > 1 else auto_category(content)
    tags = args[2:] if len(args) > 2 else []
    
    memory = {
        "id": generate_id(),
        "category": category,
        "content": content,
        "tags": tags,
        "created_at": datetime.now().isoformat(),
        "updated_at": datetime.now().isoformat(),
        "is_new": True
    }
    
    save_memory(memory)
    
    # 更新索引
    index = load_index()
    if category not in index["categories"]:
        index["categories"][category] = []
    index["categories"][category].append(memory["id"])
    
    for tag in tags:
        if tag not in index["tags"]:
            index["tags"][tag] = []
        index["tags"][tag].append(memory["id"])
    
    index["last_update"] = datetime.now().isoformat()
    save_index(index)
    
    print(f"{Colors.GREEN}✓ 记忆已添加{Colors.RESET}")
    print(f"  ID: {memory['id']}")
    print(f"  分类：{memory['category']}")
    print(f"  标签：{memory['tags']}")
    print(f"  时间：{memory['created_at']}")

def search_memory(args):
    """搜索记忆"""
    init_jiyi()
    
    if len(args) < 1:
        # 交互模式
        interactive_search()
        return
    
    keyword = " ".join(args).lower()
    
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          搜索结果：{Colors.CYAN}{keyword:<40}{Colors.BLUE}{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    
    count = 0
    index = load_index()
    
    for cat, ids in index["categories"].items():
        for memory_id in ids:
            memory = load_memory(memory_id)
            if not memory:
                continue
            
            content = memory["content"].lower()
            tags = " ".join(memory["tags"]).lower()
            category = memory["category"].lower()
            
            if keyword in content or keyword in tags or keyword in category:
                count += 1
                print_memory(memory, count)
    
    if count == 0:
        print(f"{Colors.YELLOW}✗ 未找到相关记忆{Colors.RESET}")
        print()
        print(f"{Colors.CYAN}建议:{Colors.RESET}")
        print("  1. 检查关键词是否正确")
        print("  2. 尝试其他关键词")
        print("  3. 添加新记忆：jiyi add \"内容\" 分类 标签")
        print()
        
        # 询问是否添加
        response = input(f"{Colors.CYAN}是否添加为新记忆？(y/n): {Colors.RESET}")
        if response.lower() == 'y':
            content = input("输入记忆内容：")
            if content:
                category = auto_category(content)
                print(f"自动分类：{category}")
                tags_input = input("输入标签（空格分隔）：")
                tags = tags_input.split()
                add_memory([content, category] + tags)
    else:
        print(f"{Colors.GREEN}找到 {count} 条记忆{Colors.RESET}")

def interactive_search():
    """交互搜索"""
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          jiyi - 交互式搜索                             ║{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    print(f"{Colors.CYAN}输入关键词搜索（输入 q 退出）:{Colors.RESET}")
    
    while True:
        try:
            query = input(f"\n{Colors.GREEN}> {Colors.RESET}")
            if query.lower() in ['q', 'quit']:
                break
            if query:
                search_memory([query])
        except EOFError:
            break

def list_memories(args):
    """列出记忆"""
    init_jiyi()
    
    category = args[0] if len(args) > 0 else None
    
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    if category:
        print(f"{Colors.BLUE}║          记忆列表 - {Colors.CYAN}{category:<34}{Colors.BLUE}{Colors.RESET}")
    else:
        print(f"{Colors.BLUE}║          记忆列表 - 全部{Colors.BLUE}{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    
    index = load_index()
    count = 0
    
    if category:
        ids = index["categories"].get(category, [])
        for memory_id in ids:
            memory = load_memory(memory_id)
            if memory:
                count += 1
                print(f"  {memory['content']}")
                if memory.get('is_new'):
                    print(f"    {Colors.GREEN}[NEW]{Colors.RESET}")
                print()
    else:
        for cat, ids in index["categories"].items():
            print(f"{Colors.PURPLE}【{cat}】{Colors.RESET}")
            for memory_id in ids:
                memory = load_memory(memory_id)
                if memory:
                    count += 1
                    print(f"  {memory['content']}")
                    if memory.get('is_new'):
                        print(f"    {Colors.GREEN}[NEW]{Colors.RESET}")
            print()
    
    if count == 0:
        print(f"{Colors.YELLOW}✗ 没有记忆{Colors.RESET}")
    else:
        print(f"{Colors.GREEN}共 {count} 条记忆{Colors.RESET}")

def show_categories():
    """显示分类"""
    init_jiyi()
    index = load_index()
    
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          记忆分类                                      ║{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    
    for cat, ids in index["categories"].items():
        print(f"  {Colors.PURPLE}{cat:<15}{Colors.RESET} {len(ids)} 条记忆")
    
    print()
    print(f"总计：{len(index['categories'])} 个分类")

def show_tags():
    """显示标签"""
    init_jiyi()
    index = load_index()
    
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          记忆标签                                      ║{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    
    for tag, ids in index["tags"].items():
        print(f"  {Colors.CYAN}#{tag:<15}{Colors.RESET} {len(ids)} 条记忆")
    
    print()
    print(f"总计：{len(index['tags'])} 个标签")

def show_version():
    """显示版本"""
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          jiyi - 紫微智控记忆命令                        ║{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    print(f"{Colors.CYAN}版本:{Colors.RESET} v{VERSION}")
    print(f"{Colors.CYAN}日期:{Colors.RESET} 2026-02-28")
    print(f"{Colors.CYAN}代号:{Colors.RESET} 紫微智控 - 记忆命令")
    print()
    print(f"{Colors.CYAN}功能:{Colors.RESET}")
    print("  ✅ 分门别类存储")
    print("  ✅ 最小存储空间")
    print("  ✅ 快速搜索")
    print("  ✅ 自动分类")
    print("  ✅ 新旧记忆分离")
    print()

def show_help():
    """显示帮助"""
    print(f"{Colors.BLUE}╔════════════════════════════════════════════════════════╗{Colors.RESET}")
    print(f"{Colors.BLUE}║          jiyi - 紫微智控记忆命令                        ║{Colors.RESET}")
    print(f"{Colors.BLUE}╚════════════════════════════════════════════════════════╝{Colors.RESET}")
    print()
    print(f"{Colors.CYAN}用法:{Colors.RESET}")
    print("  jiyi <命令> [参数]")
    print()
    print(f"{Colors.CYAN}命令:{Colors.RESET}")
    print("  add, a <内容> [分类] [标签...]  - 添加记忆")
    print("  search, s, find <关键词>       - 搜索记忆")
    print("  list, ls [分类]                - 列出记忆")
    print("  categories, cat                - 显示所有分类")
    print("  tags                           - 显示所有标签")
    print("  version, v                     - 显示版本")
    print("  help, h                        - 显示帮助")
    print()
    print(f"{Colors.CYAN}示例:{Colors.RESET}")
    print('  jiyi add "runtask 命令用于启动任务" 命令 runtask 自动化')
    print("  jiyi search runtask")
    print("  jiyi list 命令")
    print("  jiyi")
    print()

def print_memory(memory, index):
    """打印记忆"""
    print(f"{Colors.PURPLE}【记忆 #{index}】{Colors.RESET}")
    print(f"  内容：{memory['content']}")
    print(f"  分类：{memory['category']}")
    if memory['tags']:
        print(f"  标签：{memory['tags']}")
    print(f"  时间：{memory['created_at']}")
    if memory.get('is_new'):
        print(f"  {Colors.GREEN}[新记忆]{Colors.RESET}")
    print(f"  文件：{memory['id']}.json")
    print()

def main():
    if len(sys.argv) < 2:
        show_help()
        return
    
    command = sys.argv[1]
    args = sys.argv[2:]
    
    if command in ['add', 'a']:
        add_memory(args)
    elif command in ['search', 's', 'find']:
        search_memory(args)
    elif command in ['list', 'ls']:
        list_memories(args)
    elif command in ['categories', 'cat']:
        show_categories()
    elif command == 'tags':
        show_tags()
    elif command in ['version', 'v', '-v', '--version']:
        show_version()
    elif command in ['help', 'h', '-h', '--help']:
        show_help()
    else:
        # 默认搜索
        search_memory(sys.argv[1:])

if __name__ == "__main__":
    main()
